app/assets/javascripts/tylium/modules/image_modals.js

```diff
@@ -0,0 +1,14 @@
+document.addEventListener("turbolinks:load", function() {
+  var $modal = $('[data-behavior~=image-modal]');
+
+  $('[data-behavior~=content-textile] img').each(function() {
+    $(this).attr('data-toggle', 'modal').attr('data-target', '[data-behavior~=image-modal]');
+  });
+
+  $('[data-behavior~=content-textile] img').click(function() {
+    var title = $(this).attr('alt');
+
+    $modal.find('[data-behavior~=modal-title]').text(title);
+    $modal.find('[data-behavior~=image-modal-image]').attr('src', $(this).attr('src'));
+  });
```

Comment:
Let me see if I understand this.

1. We loop the content area of textile rendered html and assign a data attribute to images for them to target the modal, I think this also binds the functionality for the modal to open?
2. Then assign the event to the images that swaps the modal content when the image is clicked.

I'm not in love with the process. It feels like we should be able to perform the behavior in a single loop, or potentially entirely on the click function. Maybe we need to find away to append the appropriate attributes somewhere else (within textile?)

Reply:
@brianp I assume the code to add the data-attributes to the img tags would be in `lib/html/pipeline/dradis_textile_filter.rb`

I'm not sure how to add the attributes, I'm assuming we would have to patch the formatter? I chose this route since it seemed like a much simpler option but by all means, I'm open to another approach. 

I didn't see any options for this in the RedCloth gem.